Readers/src/TriggerInfoReader.cc

```diff
@@ -40,12 +89,43 @@ TriggerInfoReader::setBranchAddresses(TTree * tree)
       }
     }
   }
+  bai.setBranchAddress(ntriggerObj_, branchName_num_);
+  bai.setLenVar(64).setBranchAddress(triggerObj_id_, branchName_triggerObj_id_);
+  bai.setLenVar(64).setBranchAddress(triggerObj_filterBits_, branchName_triggerObj_filterBits_);
+  bai.setLenVar(64).setBranchAddress(triggerObj_eta_, branchName_triggerObj_eta_);
+  bai.setLenVar(64).setBranchAddress(triggerObj_phi_, branchName_triggerObj_phi_);
+  }
   return bai.getBoundBranchNames_read();
 }
 
 const TriggerInfo &
 TriggerInfoReader::read() const
 {
+  triggerInfo_.ele_trigobj_.clear();
```

Comment:
I'm not entirely convinced that this implementation is the way to go. There's no reason in my opinion to keep one `TriggerInfo` attached to the `TriggerInfoReader` class at all times. Why not remove `triggerInfo_` member and return a brand new `TriggerInfo` object every time this function is called? That's how other readers currently work. This would allow you to remove the vector clearing statements and mutable declarations from the class definition. Instead, you could just declare completely new vectors of integers (or even come up with a tiny struct called, say, `TriggerObj` that you can collect into just one vector) and pass them to the constructor of `TriggerInfo`.

Reply:
Hi Karl, the TriggerInfo object is a data-member of the TriggerInfoReader also in the master branch. I have written the code like such to avoid that the TriggerInfo object needs to be passed by copy when the EventReader class calls the TriggerInfoReader::read() function (I prefer to avoid copying the TriggerInfo object, because it is a quite complex object). I don't feel strongly about this feature though. If you prefer that the TriggerInfoReader::read() function returns a TriggerInfo object by copy, then I propose that I change the code accordingly after Saswati's PR has been merged into the master branch.

Reply:
Hi Christian, I prefer my proposal for the reasons I already gave. Other reader classes return objects by copy all the time, and they construct much more complex objects than `TriggerInfo` here. I'm totally fine if you change the code after this PR is merged, which is quite long already.

Reply:
Hi Karl, 

I took another look at the TriggerInfoReader and TriggerInfo code today and I found a strong motivation (in my opinion) to keep the code as it is, i.e. keep a TriggerInfo object as private data-member of the TriggerInfoReader class and return a reference to the private data-member in the TriggerInfoReader::read() method. 

The reason is that the TriggerInfo object is not only initialized from the nanoAOD, but also from the python config
  https://github.com/HEP-KBFI/TallinnNtupleProducer/blob/main/Readers/python/trigger_2016_mc_cfi.py

The logic of the TriggerInfo+TriggerInfoReader class is similar to that of the EventInfo+EventInfoReader (and AnalysisConfig) class (so the logic is not unprecedented).

I would prefer to keep the TriggerInfoReader and TriggerInfo code as it is for now.